nidx/nidx_paragraph/src/query_parser/keyword_parser.rs

```diff
@@ -0,0 +1,189 @@
+// Copyright (C) 2021 Bosutech XXI S.L.
+//
+// nucliadb is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at info@nuclia.com.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+use tantivy::Term;
+use tantivy::query::{AllQuery, BooleanQuery, Occur, PhraseQuery, Query, TermQuery};
+use tantivy::schema::IndexRecordOption;
+
+use crate::ParagraphSchema;
+
+use super::tokenizer::Token;
+
+/// Convert a tokenized query into a tantivy keyword query
+///
+/// Empty queries will match everything.
+pub fn parse_keyword_query<'a>(query: &'a [Token<'a>], schema: &ParagraphSchema) -> Box<dyn Query> {
+    let mut subqueries = vec![];
+    for item in query {
+        match item {
+            Token::Literal(literal) => {
+                let q: Box<dyn Query> = Box::new(TermQuery::new(
+                    Term::from_field_text(schema.text, literal),
+                    IndexRecordOption::Basic,
+                ));
+                subqueries.push((Occur::Should, q));
+            }
+            Token::Quoted(quoted) => {
+                let mut terms: Vec<Term> = quoted
```

Comment:
I was wondering if this part of the code can be shared with the fuzzy parser? If I'm reading correctly, Literals are different for keyword/fuzzy, but phrase and excluded looks like they are the same?